File: src/data/types/scannerCandidate.ts
import type { TierColor } from "./tierColor.ts";

// Necessary correction to the Scanner/Probe GDD's own literal pseudocode,
// same precedent as Phase 4's CrewCandidate/CrewMember split and Phase 5's
// ShipCandidate/Ship split: ScannerPool.availableScanners was written as
// Scanner[] in the GDD's own pseudocode, but an unpurchased pool entry has
// no real ownerId yet -- Scanner.ownerId is required and meaningless before
// a purchase transfers ownership. ScannerCandidate omits only that field;
// purchaseScanner() (Agent 20) is what turns one into a real Scanner.
export interface ScannerCandidate {
  id: string;
  tier: TierColor;
}

File: unity/ProfitableCore/Schema/ScannerCandidate.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/scannerCandidate.ts. Omits Scanner.OwnerId only --
// an unpurchased pool entry has no real owner yet.
public class ScannerCandidate
{
    public string Id { get; set; } = string.Empty;
    public TierColor Tier { get; set; }
}
